### DISCUSSION TITLE
Can not change the default guard to `sanctum`

### DISCUSSION THREAD

**Discussion by sohailahmad07** [spatie/laravel-permission#2777] [Q&A]
Hi 

i am using laravel-permission with laravel jetstream and jetstream use `auth:sanctum` in the web.php but the default guard in laravel permission is web and any time i tried to add a permission to role i get the error 

```
There is no permission named `can:view-users` for guard `sanctum`. 
```

i have tried to change the default guard as instructed by the doc but it still create role and permission with `web` guard
 
 ```php
 
 // config/auth.php
  'defaults' => [
        'guard' => env('AUTH_GUARD', 'web'),
        'passwords' => env('AUTH_PASSWORD_BROKER', 'users'),
    ],
    
 'guards' => [
        'sanctum' => [
            'driver' => 'sanctum',
            'provider' => 'users',
        ],
        'web' => [
            'driver' => 'session',
            'provider' => 'users',
        ],
    ],
```    
```php
// User.php

 protected function getDefaultGuardName(): string
    {
        return 'sanctum';
    }
 ```

**Comment by erikn69** [comment]
On laravel 11, maybe try https://github.com/spatie/laravel-permission/issues/2729#issuecomment-2465716861
